File: DataLayers/sp_pid/SPPIDDataLayer/SPPIDProvider.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.IO

Imports org.iringtools.adapter
Imports org.iringtools.utility
Imports org.iringtools.library
Public Class SPPIDProvider

    Implements IDisposable
    Private _settings As AdapterSettings = Nothing
    Private _configurationPath As String = String.Empty
    Private _configuration As SPPIDConfiguration = Nothing

    Public Sub New(configuration As SPPIDConfiguration)
        InitializeProvider(configuration)
    End Sub
    Public Sub New(settings As AdapterSettings)
        _settings = settings
        _configurationPath = Path.Combine(_settings("AppDataPath"), "SPPID-configuration." + _settings("Scope") & ".xml")

        If File.Exists(_configurationPath) Then

            'if (_configuration.Generate)
            '{
            ' _configuration = ProcessConfiguration(_configuration, null);
            '  _configuration.Generate = false;
            '  Utility.Write<SPPIDConfiguration>(_configuration, _configurationPath, true);
            ' }
            InitializeProvider(utility.Utility.Read(Of SPPIDConfiguration)(_configurationPath))
        End If
    End Sub

    Public Sub InitializeProvider(configuration As SPPIDConfiguration)
        If configuration IsNot Nothing Then
            'if (File.Exists(_configuration.Location))
            '{
            ' // if (_stream == null) _stream = OpenStream(_configuration.Location);
            '  if (_document == null) _document = GetDocument(_configuration.Location);
            '  if (_configuration.Generate)
            '  {
            '    _configuration = ProcessConfiguration(_configuration, null);
            '    _configuration.Generate = false;
            '    Utility.Write<SPPIDConfiguration>(_configuration, _configurationPath, true);
            '  }
            '}
            _configuration = configuration
        End If
    End Sub

    Public Overloads Sub Dispose() Implements IDisposable.Dispose
      
    End Sub


End Class

File: Libraries/SPPID/SPPIDProvider.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO;

using org.iringtools.adapter;
using org.iringtools.utility;
using org.iringtools.library;

namespace org.iringtools.adapter.datalayer
{
    class SPPIDProvider : IDisposable
    {
        private AdapterSettings _settings = null;
        private string _configurationPath = string.Empty;
        private SPPIDConfiguration _configuration = null;

        public SPPIDProvider(SPPIDConfiguration configuration)
        {
            InitializeProvider(configuration);
        }
        public SPPIDProvider(AdapterSettings settings)
        {
            _settings = settings;
            _configurationPath = Path.Combine(_settings["AppDataPath"], "SPPID-configuration." + _settings["Scope"] + ".xml");

            if (File.Exists(_configurationPath))
            {
                InitializeProvider(Utility.Read<SPPIDConfiguration>(_configurationPath));

                //if (_configuration.Generate)
                //{
                // _configuration = ProcessConfiguration(_configuration, null);
                //  _configuration.Generate = false;
                //  Utility.Write<SPPIDConfiguration>(_configuration, _configurationPath, true);
                // }
            }
        }

        public void InitializeProvider(SPPIDConfiguration configuration)
        {
            if (configuration != null)
            {
                _configuration = configuration;
                //if (File.Exists(_configuration.Location))
                //{
                // // if (_stream == null) _stream = OpenStream(_configuration.Location);
                //  if (_document == null) _document = GetDocument(_configuration.Location);
                //  if (_configuration.Generate)
                //  {
                //    _configuration = ProcessConfiguration(_configuration, null);
                //    _configuration.Generate = false;
                //    Utility.Write<SPPIDConfiguration>(_configuration, _configurationPath, true);
                //  }
                //}
            }
        }

        public void Dispose()
        {
            
        }

    }
}
